### ISSUE TITLE
is developing extensions for g46/45 worthy?

### ISSUE THREAD

**Issue by qwreey** [qwreey/quick-settings-tweaks#136]
The latest version of Gnome is horribly unstable, even I can't develop with it.
It crashes even if no extension is enabled.

It is hard to develop gnome extension with gnome 46/45. Every time I reload gnome-shell, it just crashes. unlike g44, it requires 'logout and login' which is horribly slower than just reloading gnome-shell.

It makes extension development very slow, even, I can't follow gnome-shell updates.
We can use 'nested gnome-shell', yes, I can. but that one is limited. I can't test every feature with it

Recently, I think, the gnome team tried to kill extension developers. They made a bunches of breaking changes
They can retain options to use 'imports' global variable rather than force to use ESM.

Gnome developers must care about the 'extension environment' because almost everyone uses.
In other words, Gnome is alive with an extension ecosystem.

I don't know what's happening there because I have a busy life. I don't have a bunch of time.

**Comment by qwreey** [comment]
![20240331_171103.jpg](https://github.com/qwreey/quick-settings-tweaks/assets/46598063/19ffb7fc-ffb6-4e82-9bf2-97f6ac5a7578)

I see thus screen hourly

**Comment by qwreey** [comment]
![20240331_190428.jpg](https://github.com/qwreey/quick-settings-tweaks/assets/46598063/ecb4ec28-9bca-4c31-93e4-e6289fb7c4f0)

wow.

**Comment by qwreey** [comment]
![20240331_191315.jpg](https://github.com/qwreey/quick-settings-tweaks/assets/46598063/f8e3c173-dd1d-4c88-bfbc-5b07b5d3aeda)

One more time.

I will move to awesome-wm. at this moment. i will wash my hands on this project.

I will be back if Gnome is working fine

**Comment by qwreey** [comment]
As im in college, I have to time to continue. I need a de system that is stable enough..
Gnome is not.

**Comment by qwreey** [comment]
Sorry for everyone.

**Comment by Ienien77** [comment]
I've also noticed those instabilities but only with version 46 which was updated recently on my machine. It was working great on 45. No idea what cause extentions to not work anymore on  gnome 46. I wish i was able to understand coding stuff...

**Comment by qwreey** [comment]
If, looking-glass allows import syntax, have better UX (like gtk4/libadwaita inspect) and the gnome has better development experience. (such as fast reload, which allows continuous developing, clear error output, and logging/debugging system)

I will spend my time for gnome, even if i have a busy life. (I have many prototypes, but, discontinued for the same reason)

Development experience is very important. Not only does it affect extension developers, But also it affects themselves.

Gnome developers should understand how 'continuous development experience' is important.

How do you think if 'hot reload' features are gone from web-builders? Maybe you will have a horribly slower development speed.

I love gnome-shell. I wrote almost 12k lines of code for gnome. (including my private extensions)

But, at this time. it's a time blackhole.
Another de/wm?
awesome: Just one keyboard shortcut for reloading config(code) files.

I wasn't disappointed with instability. Always, has been. And instability is common on open-source softwares.

The main reason is how stupid it is inefficient.
Not only focus on many features and it is modern (the result is), They should focus 'How to develop effectively'

I'm tired of the very low-level development environment (even though it uses JavaScript).

**Comment by qwreey** [comment]
VM is a good option for creating an effective development environment, but that's very heavy..

Currently, I try to use the 'container'

container has better GPU support for many situations, lightweight, and Hot-reloadable.

Also, has no side effects on the host, and we can just mount working-dir to inside (which allows automated building)

**Comment by ZimbiX** [comment]
A container's a good idea. What's your setup? I came across this [great set of blog posts covering containerised automated testing for Gnome extensions](https://reddit.com/r/gnome/comments/t5lxkm/enable_continuous_integration_tests_for_your/) a while ago, but haven't given it a go yet. I think that'd need VNC or something added to be able to do development with
